cmd/acr/purge.go

```diff
@@ -4,27 +4,222 @@
 package main
 
 import (
+	"context"
 	"fmt"
 	"io"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
 
-	acrapi "github.com/Azure/libacr/golang"
+	"github.com/AzureCR/acr-cli/cmd/api"
 	"github.com/spf13/cobra"
 )
 
 const (
-	purgeLongMessage = `` // TODO
+	purgeLongMessage = `acr purge: untag old images and delete dangling manifests.`
+	exampleMessage   = `
+Delete all tags that are older than 1 day
+  acr purge -r MyRegistry --repository MyRepository --ago 1d
+
+Delete all tags that are older than 1 day and begin with hello
+  acr purge -r MyRegistry --repository MyRepository --ago 1d --filter "^hello.*"
+
+Delete all dangling manifests
+  acr purge -r MyRegistry --repository MyRepository --dangling`
+)
+
+var (
+	registryName string
+	username     string
+	password     string
+	ago          string
+	dangling     bool
+	filter       string
+	repoName     string
 )
 
 func newPurgeCmd(out io.Writer) *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "purge",
-		Short: "", // TODO
-		Long:  purgeLongMessage,
+		Use:     "purge",
+		Short:   "Delete images from a registry.",
+		Long:    purgeLongMessage,
+		Example: exampleMessage,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			fmt.Println("The default base uri is:", acrapi.DefaultBaseURI)
+			ctx := context.Background()
+			var e error
+			var wg sync.WaitGroup
```

Comment:
Could you explain why this wait group `wg` is required to be passed to the following functions instead of declaring it locally inside of those functions. Is it shared between `PurgeTags` and `PurgeDanglingManifests`?

Reply:
In a previous version it was shared, but in this one it is not so both functions can have a local WaitGroup instead